### PR TITLE
Local and Test environment for Docker based submissions

### PR DESCRIPTION
**Pull request by vkartik97**
Deliverables:
- Separated Production(Release) and Test/Dev Environments
- Added support for  Local Docker Registry to push images for Docker based Challenges.
   Default URL for registry is `localhost:5000`. Ref: [registry:2](https://hub.docker.com/_/registry)
- Added Base Test for `$ evalai push`. The test involves 
   - `test_make_submission_for_docker_based_challenge_setup`: Spawning a new *isolated* `registry:2` on random open port(`get_open_port`), where image push will be tested
   - `test_make_submission_for_docker_based_challenge`: Push a `Scratch` image to registry created in `setup`.
   - `test_make_submission_for_docker_based_challenge_teardown`: To delete the spawned registry after the test is done.

### CODE REVIEW

**Comment by vkartik97** [comment]
@deshraj @RishabhJain2018 @rishabhbhardwaj @fristonio Please review

**Comment by deshraj** [comment]
@vkartik97 can you please add a short description to this PR?

**Comment by vkartik97** [comment]
@deshraj Added :)

File: pytest.ini [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[pytest]
+env =
+    EVALAI_CLI_ENVIRONMENT=TEST
```
Comment (RishabhJain2018) [inline pytest.ini]: Please add a new line here.

File: tests/test_submissions.py [hunk 2]
```diff
@@ -177,3 +237,57 @@ def test_make_submission_when_file_is_valid_with_metadata(self):
...
+                push,
+                [
+                    image_tag,
+                    "-p",
+                    "philip-phase-2019",
+                    "-u",
+                    "localhost:{0}".format(registry_port),
+                ],
+            )
+            print(result.output.strip())
```
Comment (RishabhJain2018) [inline tests/test_submissions.py]: Please remove this print statement.

**Comment by RishabhJain2018** [comment]
@vkartik97 Can you please write the usage of `evalai push` command in the development version?

File: evalai/submissions.py [hunk 3]
```diff
@@ -114,32 +127,44 @@ def push(image, phase):
...
     repository_uri = response["success"]["docker_repository_uri"]
 
-    AWS_ACCOUNT_ID = federated_user["FederatedUser"]["FederatedUserId"].split(
-        ":"
-    )[0]
-    AWS_SERVER_PUBLIC_KEY = federated_user["Credentials"]["AccessKeyId"]
-    AWS_SERVER_SECRET_KEY = federated_user["Credentials"]["SecretAccessKey"]
-    SESSION_TOKEN = federated_user["Credentials"]["SessionToken"]
+    # Default Environment
+    if ENVIRONMENT == "PRODUCTION":
```
Comment (RishabhJain2018) [inline evalai/submissions.py]: This `if` statement will come up after line 122, no?

File: evalai/submissions.py [hunk 4]
```diff
@@ -114,32 +127,44 @@ def push(image, phase):
     federated_user = response["success"]["federated_user"]
     repository_uri = response["success"]["docker_repository_uri"]
 
-    AWS_ACCOUNT_ID = federated_user["FederatedUser"]["FederatedUserId"].split(
-        ":"
-    )[0]
-    AWS_SERVER_PUBLIC_KEY = federated_user["Credentials"]["AccessKeyId"]
-    AWS_SERVER_SECRET_KEY = federated_user["Credentials"]["SecretAccessKey"]
-    SESSION_TOKEN = federated_user["Credentials"]["SessionToken"]
+    # Default Environment
```
Comment (RishabhJain2018) [inline evalai/submissions.py]: Please change this to  `Production Environment`

File: evalai/submissions.py (hunk 3 above)
Comment (RishabhJain2018) [reply evalai/submissions.py]: Please ignore this, I think if we will merge this one, then we won't need this change: https://github.com/Cloud-CV/EvalAI/pull/2359/files

File: pytest.ini (hunk 1 above)
Comment (vkartik97) [reply pytest.ini]: Done

File: tests/test_submissions.py (hunk 2 above)
Comment (vkartik97) [reply tests/test_submissions.py]: Done

File: evalai/submissions.py (hunk 4 above)
Comment (vkartik97) [reply evalai/submissions.py]: Done

**Pull request merged**